.gitignore

```diff
@@ -0,0 +1,150 @@
+# Created by https://www.gitignore.io/api/java,maven,windows,intellij
```

Comment:
Just noticed there's another PR open for adding a gitignore file. Can you remove this file from this PR and we'll just get the other PR merged and you can pull it into this branch.